### DISCUSSION TITLE
Grouping log lines: set the folded/unfolded state with `::endgroup::` at the end of the logs

### DISCUSSION THREAD

**Discussion by MisterDA** [community/community#39411] [Actions]
### Select Topic Area

Product Feedback

### Body

Hello all,

I'm contributing to a test framework. I've added support for it to detect when it's running in GitHub Actions, using the `CI` and `GITHUB_ACTIONS` environment variables. Instead of simply writing the test results, the test framework will also output [Workflow commands for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions) to get nice annotations in the GHA code, eventually linking back to the code where the error was raised, etc.

The tests are grouped in test suites. I've had the idea of grouping each test suite in a fold 
```
::group::{title}
::endgroup::
```
for the user to be able to navigate more easily between them. The test framework doesn't buffer the logs of each test, so they are just streamed, and the result of the test is only known when it's finished so _importantly_ I don't know whether the test has succeeded or failed before I start printing its logs.

I would like to be able to set the folded/unfolded state of each group _when the group ends_: I the test suite succeeded then its group can remain folded, if the test suite failed, I can set its group unfolded so that the user will see the failures right away.
This feature would be useful on the closing tag because we might not know when opening the group the final folded/unfolded state.

If folded is the default state, this could be as easy as this, with `::endgroup::` taking precedence.
```
::group::{title,unfold}
::endgroup::{unfold}
```

Note that from the test framework, I cannot use GHA js API, I can only output these annotations in the logs.

Thank you!

**Comment by lengau** [comment]
This would make a big difference to us, as we want to run all linters but not collapse the group if the linter fails.
